manager/scheduler/filter.go

```diff
@@ -217,3 +217,46 @@ func (f *ConstraintFilter) Explain(nodes int) string {
 	}
 	return fmt.Sprintf("scheduling constraints not satisfied on %d nodes", nodes)
 }
+
+// PlatformFilter selects only nodes that run the required platform.
+type PlatformFilter struct {
+	supportedPlatforms []*api.Platform
+}
+
+// SetTask returns true when the filter is enabled for a given task.
+func (f *PlatformFilter) SetTask(t *api.Task) bool {
+	c := t.Spec.GetContainer()
+	if c != nil {
+		if len(c.SupportedPlatforms) > 0 {
+			// copy the platform information
+			f.supportedPlatforms = c.SupportedPlatforms
+			return true
+		}
+	}
+	return false
+}
+
+// Check returns true if the task can be scheduled into the given node.
+func (f *PlatformFilter) Check(n *NodeInfo) bool {
+	// if the supportedPlatforms field is empty, then either it wasn't
+	// provided or there are no constraints
+	if len(f.supportedPlatforms) == 0 {
+		return true
+	}
+	// check if the platform for the node is supported
+	nodePlatform := n.Description.Platform
+	for _, p := range f.supportedPlatforms {
+		if p.Architecture == nodePlatform.Architecture && p.OS == nodePlatform.OS {
```

Comment:
The node info from `docker info` contains `Operating System` and `OSType`. I think currently we are collecting only one of them in NodeInfo. I think both are useful. 

```
dchen@vm2:~$ docker info | egrep "Operating|OS"
Operating System: Ubuntu 14.04.3 LTS
OSType: linux
```